airbyte-ci/connectors/pipelines/pipelines/helpers/utils.py

```diff
@@ -101,13 +101,10 @@ async def get_file_contents(container: Container, path: str) -> Optional[str]:
     Returns:
         Optional[str]: The file content if the file exists in the container, None otherwise.
     """
-    try:
-        return await container.file(path).contents()
-    except QueryError as e:
-        if "no such file or directory" not in str(e):
-            # this error could come from a network issue
-            raise
-    return None
+    dir_name, file_name = os.path.split(path)
+    if file_name not in set(await container.directory(dir_name).entries()):
+        return None
+    return await container.file(path).contents()
```

Comment:
The precondition here is that `path` is a file path and not a dir path, I'm not sure its enforced but I'm fine with it, Dagger will raise an exception if the path leads to a directory.

Reply:
I think that raising an exception is fine in that case, `get_file_contents` really makes no sense if applied to a directory.